app/gui2/e2e/widgets.spec.ts

```diff
@@ -0,0 +1,247 @@
+  await mockMethodCallInfo(
+    page,
+    {
+      binding: 'aggregated',
+      expr: 'Aggregate_Column.Group_By',
+    },
+    {
+      methodPointer: {
+        module: 'Standard.Table.Data.Aggregate_Column',
+        definedOnType: 'Standard.Table.Data.Aggregate_Column.Aggregate_Column',
+        name: 'Group_By',
+      },
+      notAppliedArguments: [0, 1],
+    },
+  )
+
+  // Change aggregation type
+  const firstItem = columnsArg.locator('.item > .WidgetPort > .WidgetSelection')
+  await firstItem.click()
+  await dropDown.expectVisibleWithOptions(page, ['Group By', 'Count', 'Count Distinct'])
+  await dropDown.clickOption(page, 'Count Distinct')
+  await expect(columnsArg.locator('.WidgetToken')).toHaveText([
+    'Aggregate_Column',
+    '.',
+    'Count_Distinct',
+  ])
+  await mockMethodCallInfo(
+    page,
+    {
+      binding: 'aggregated',
+      expr: 'Aggregate_Column.Count_Distinct',
+    },
+    {
+      methodPointer: {
+        module: 'Standard.Table.Data.Aggregate_Column',
+        definedOnType: 'Standard.Table.Data.Aggregate_Column.Aggregate_Column',
+        name: 'Count_Distinct',
+      },
+      notAppliedArguments: [0, 1, 2],
+    },
+  )
+
+  // Set column
+  const columnArg = firstItem.locator('.WidgetSelection').first()
+  await columnArg.click()
+  await dropDown.expectVisibleWithOptions(page, ['column 1', 'column 2'])
+  await dropDown.clickOption(page, 'column 1')
+  await expect(columnsArg.locator('.WidgetToken')).toHaveText([
+    'Aggregate_Column',
+    '.',
+    'Count_Distinct',
+    '"',
+    'column 1',
+    '"',
+  ])
+
+  // Add another aggregate
+  await columnsArg.locator('.add-item').click()
+  await expect(columnsArg.locator('.WidgetToken')).toHaveText([
+    'Aggregate_Column',
+    '.',
+    'Count_Distinct',
+    '"',
+    'column 1',
+    '"',
+    'Aggregate_Column',
+    '.',
+    'Group_By',
+  ])
+  await mockMethodCallInfo(
+    page,
+    {
+      binding: 'aggregated',
+      expr: 'Aggregate_Column.Group_By',
+    },
+    {
+      methodPointer: {
+        module: 'Standard.Table.Data.Aggregate_Column',
+        definedOnType: 'Standard.Table.Data.Aggregate_Column.Aggregate_Column',
+        name: 'Group_By',
+      },
+      notAppliedArguments: [0, 1],
+    },
+  )
+
+  // Set new aggregate's column
+  const secondItem = columnsArg.locator('.item > .WidgetPort > .WidgetSelection').nth(1)
+  const secondColumnArg = secondItem.locator('.WidgetSelection').first()
+  await secondColumnArg.click()
+  await dropDown.expectVisibleWithOptions(page, ['column 1', 'column 2'])
+  await dropDown.clickOption(page, 'column 2')
+  await expect(secondItem.locator('.WidgetToken')).toHaveText([
+    'Aggregate_Column',
+    '.',
+    'Group_By',
+    '"',
+    'column 2',
+    '"',
+  ])
+
+  // Switch aggregates
+  //TODO[ao] I have no idea how to emulate drag. Simple dragTo does not work (some element seem to capture event).
```

Comment:
Should we link this TODO to a proper issue?

Reply:
I added points in the e2e test task.